phantazm-mob/src/main/java/com/github/phantazmnetwork/mob/config/MobModelConfigProcessor.java

```diff
@@ -0,0 +1,165 @@
+            }
+
+            triggers.put(triggerKey, List.copyOf(skills));
+        }
+
+        Component displayName = componentProcessor.dataFromElement(element.getElementOrThrow("displayName"));
+
+        ConfigNode equipmentNode = element.getNodeOrThrow("equipment");
+        Map<EquipmentSlot, ItemStack> equipment = new HashMap<>(equipmentNode.size());
+        for (Map.Entry<String, ConfigElement> entry : equipmentNode.entrySet()) {
+            EquipmentSlot equipmentSlot = EQUIPMENT_SLOT_PROCESSOR.dataFromElement(new ConfigPrimitive(entry.getKey().toUpperCase()));
+            equipment.put(equipmentSlot, itemStackProcessor.dataFromElement(entry.getValue()));
+        }
+
+        float maxHealth = element.getNumberOrThrow("maxHealth").floatValue();
+
+        return new MobModel(key, descriptor, goalGroups, triggers, displayName, equipment, maxHealth);
+    }
+
+    @Override
+    public @NotNull ConfigElement elementFromData(@NotNull MobModel model) throws ConfigProcessException {
+        ConfigElement key = KEY_PROCESSOR.elementFromData(model.key());
+
+        ConfigElement descriptor = descriptorProcessor.elementFromData(model.getDescriptor());
+
+        ConfigList goalGroupsList = new ArrayConfigList(model.getGoalGroups().size());
+        for (Collection<Goal> group : model.getGoalGroups()) {
+            ConfigList goalGroupList = new ArrayConfigList(group.size());
+            for (Goal goal : group) {
+                goalGroupList.add(goalProcessor.elementFromData(goal));
+            }
+
+            goalGroupsList.add(goalGroupList);
+        }
+
+        ConfigList triggers = new ArrayConfigList(model.getTriggers().size());
+        for (Map.Entry<Key, Collection<Skill>> trigger : model.getTriggers().entrySet()) {
+            ConfigList skillList = new ArrayConfigList(trigger.getValue().size());
+            for (Skill skill : trigger.getValue()) {
+                skillList.add(skillProcessor.elementFromData(skill));
+            }
+
+            ConfigNode node = new LinkedConfigNode(2);
+            node.put("key", KEY_PROCESSOR.elementFromData(trigger.getKey()));
+            node.put("skills", skillList);
+            triggers.add(node);
+        }
+
+        Optional<Component> displayName = model.getDisplayName();
+        ConfigElement displayNameElement;
+        if (displayName.isPresent()) {
+            displayNameElement = componentProcessor.elementFromData(displayName.get());
+        }
+        else {
+            displayNameElement = new ConfigPrimitive(null);
+        }
+
+        ConfigNode equipmentNode = new LinkedConfigNode(model.getEquipment().entrySet().size());
+        for (Map.Entry<EquipmentSlot, ItemStack> entry : model.getEquipment().entrySet()) {
+            ConfigElement slotElement = EQUIPMENT_SLOT_PROCESSOR.elementFromData(entry.getKey());
+            if (!slotElement.isString()) {
+                throw new ConfigProcessException("equipment slot processor did not create a string");
```

Comment:
enumProcessor is guaranteed to give you a cfg processor that returns a ConfigPrimitive holding a string, so this check is unnecessary, but not bad if you're going for an abundance of caution

Reply:
in case of the future lol